### DISCUSSION TITLE
parse_file_as equivalent in v2

### DISCUSSION THREAD

**Discussion by NunchakusLei** [pydantic/pydantic#8568] [Question] [answered]
Looks like v2 deprecated the `parse_file_as` function. I usually use it to parse a list of defined Pydantic model from JSON like the following,

```python
parse_file_as(List[MyPydanticModel], <path_to_the_file>)
```

Does anyone know how to implement the same or similar function in v2? 

Similar question apply to parse_obj_as and parse_raw_as, too.

**Comment by qerub** [comment, marked as answer]
I ended up doing this:

```python
with open(path_to_the_file) as f:
    pydantic.TypeAdapter(List[MyPydanticModel]).validate_json(f.read())
```

Improvement suggestions are welcome! :)

**Reply by flickerfly** [reply]
Works good for me! Thanks for dropping a solution here.
